src/app/api/core/utils/withRetry.ts

```diff
@@ -0,0 +1,49 @@
+import { StatusableError } from '@/types/CopilotApiError'
+import pRetry, { FailedAttemptError } from 'p-retry'
+import * as Sentry from '@sentry/nextjs'
+
+export const withRetry = async <T>(fn: (...args: any[]) => Promise<T>, args: any[]): Promise<T> => {
+  let isEventProcessorRegisterd = false
```

Comment:
I think that setting `isEventProcessorRegisterd` to true after adding the event processor ensures that we don’t re-register the event processor. However, there's a potential edge case where, if the scope is not reused or gets cleared between retries, the processor may not be registered again.

Reply:
This scope should only exist after withRetry is triggered and the child exception is caught. So it not being registered again will not be an issue.